### PR TITLE
feat(node): support 'npm' as a package manager

### LINKED ISSUE #156: Feature request: Use NPM instead of Yarn
Yarn is not playing well with private dependencies from Github Package Registry. It would be nice to have an option to choose between npm and yarn, like in create-react-app: https://create-react-app.dev/docs/getting-started/#selecting-a-package-manager

### PR DESCRIPTION
**Pull request by RafalWilinski**
Adds support for specifying `packageManager` for node projects (and derivatives). The default is still `yarn` but `npm` is also supported.

Closes #156

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by eladb — CHANGES_REQUESTED** [review]
See notes

File: src/node-project.ts [hunk 1]
```diff
@@ -447,6 +454,10 @@ export class NodeProject extends Project {
 
   protected readonly npmRegistry: string;
 
+  public readonly useNpm: boolean;
+
+  public readonly scriptRunCmd: string;
```
[the remark this answers is not in the document]
Comment (RafalWilinski) [reply src/node-project.ts:459]: I needs to be public in order to be used inside e.g `NodeBuildWorkflow`

**Comment by RafalWilinski** [comment]
`useNpm` changed to `NodePackageManager` enum. Let me know if there's anything else.

**Review by eladb — APPROVED** [review]
Made a few tweaks.

**Pull request merged**